com.unity.shadergraph/Editor/Drawing/Inspector/GraphDataPropertyDrawer.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Reflection;
+using UnityEditor.ShaderGraph;
+using UnityEditor.ShaderGraph.Drawing;
+using UnityEditor.ShaderGraph.Internal;
+using UnityEngine.UIElements;
+
+namespace Drawing.Inspector
+{
+    [SGPropertyDrawer(typeof(GraphData))]
+    public class GraphDataPropertyDrawer : IPropertyDrawer
+    {
+        public delegate void ChangeConcretePrecisionCallback(ConcretePrecision newValue);
+        public delegate void PostTargetSettingsChangedCallback();
+
+        private PostTargetSettingsChangedCallback m_postChangeTargetSettingsCallback;
+        private ChangeConcretePrecisionCallback m_postChangeConcretePrecisionCallback;
+
+        // Targets have persistent data that needs to be maintained between interactions
+        // Hence we are storing a reference to this particular property drawer to continue using it
+        // private TargetPropertyDrawer m_TargetPropertyDrawer = new TargetPropertyDrawer();
+
+        //Dictionary<Target, bool> m_TargetFoldouts = new Dictionary<Target, bool>();
+
+        public void GetPropertyData(
+            PostTargetSettingsChangedCallback postChangeValueCallback,
+            ChangeConcretePrecisionCallback changeConcretePrecisionCallback)
+        {
+            m_postChangeTargetSettingsCallback = postChangeValueCallback;
+            m_postChangeConcretePrecisionCallback = changeConcretePrecisionCallback;
+        }
+
+        internal VisualElement CreateGUI(GraphData graphData)
+        {
+            if (graphData == null)
+                throw new InvalidCastException(
```

Comment:
We should try to avoid throwing since it puts the graph into the Soft Lock state. You can write an Editor.log message but probably just return an empty property sheet in this case so it doesn't break the user.